### PR TITLE
Add model01 and doio/kb16 keymaps

### PR DESCRIPTION
**Pull request by pugmajere**
This includes two new personal keymaps:
- a basic combo keymap for MacOS and Linux, for interacting with Google Meet using the doio/kb16
- a port of a keyboardio/model01 keymap from Kaleidoscope to qmk, tweaking a bunch of things to make it more reliable. (Fewer LED modes, adding debouncing to deal with early model hardware problems.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* None

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/doio/kb16/keymaps/pugmajere/keymap.c [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+// OLED animation
+#include "lib/layer_status/layer_status.h"
+
+// Each layer gets a name for readability, which is then used in the keymap matrix below.
+// The underscores don't mean anything - you can have a layer called STUFF or any other name.
+// Layer names don't all need to be of the same length, obviously, and you can also skip them
+// entirely and just use numbers.
+
+enum layer_names { _BASE, _FN, _FN1, _FN2 };
```
Comment (drashna) [inline keyboards/doio/kb16/keymaps/pugmajere/keymap.c:29]:
```suggestion
enum layer_names { 
    _BASE, 
    _FN, 
    _FN1, 
    _FN2,
};
```

File: keyboards/keyboardio/model01/keymaps/pugmajere/config.h [hunk 2]
```diff
@@ -0,0 +1,51 @@
+#pragma once
```
Comment (drashna) [inline keyboards/keyboardio/model01/keymaps/pugmajere/config.h:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/keyboardio/model01/keymaps/pugmajere/keymap.c [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+  )
+*/
+// clang-format on
+
+void mod_type(uint16_t modcode, uint16_t keycode) {
+    register_mods(MOD_BIT(modcode));
+    register_code(keycode);
+    unregister_code(keycode);
+    unregister_mods(MOD_BIT(modcode));
+}
```
Comment (drashna) [inline keyboards/keyboardio/model01/keymaps/pugmajere/keymap.c:86]: `tap_code16` does this, explicitly.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/keyboardio/model01/keymaps/pugmajere/keymap.c (hunk 3 above)
Comment (pugmajere) [reply keyboards/keyboardio/model01/keymaps/pugmajere/keymap.c:86]: Oh! Thanks, that is a bit of code I didn't really want to have in here, I just didn't find tap_code16 when I was puzzling this out.

File: keyboards/keyboardio/model01/keymaps/pugmajere/config.h (hunk 2 above)
Comment (pugmajere) [reply keyboards/keyboardio/model01/keymaps/pugmajere/config.h:1]: Done.

**Comment by pugmajere** [comment]
I think I've got everything fixed now; thanks for the review!

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**